osu.Game/Online/Chat/ChannelManager.cs

```diff
@@ -99,6 +102,28 @@ private void currentChannelChanged(ValueChangedEvent<Channel> e)
         /// </summary>
         private readonly Queue<Action> postQueue = new Queue<Action>();
 
+        /// <summary>
+        /// Scans for inline commands of the form $cmd and returns the newly processed text
+        /// </summary>
+        public string ProcessInlineCommands(string text) {
```

Comment:
this is in the wrong place. it should likely be its own class with tests for specific replacements. `ChannelManager` should not be responsible for chat commands to this level.